lib/inventory/inventory_api.rb

```diff
@@ -105,47 +100,49 @@ def get_config_data_for_resource(resource_id, feed)
     # Obtain the child resources of the passed resource. In case of a WildFly server,
     # those would be Datasources, Deployments and so on.
     # @param [Resource] parent_resource Resource to obtain children from
+    # @param [Boolean] recursive Whether to fetch also all the children of children of ...
     # @return [Array<Resource>] List of resources that are children of the given parent resource.
     #   Can be empty
-    def list_child_resources(parent_resource)
+    def list_child_resources(parent_resource, recursive = false)
       the_feed = hawk_escape parent_resource.feed
       the_id = hawk_escape parent_resource.id
 
       ret = http_get('/feeds/' + the_feed +
-                     '/resources/' + the_id + '/children')
-      val = []
-      ret.each { |r| val.push(Resource.new(r)) }
-      val
+                     '/resources/' + the_id +
+                     (recursive ? '/recursiveChildren' : '/children'))
+      ret.map { |r| Resource.new(r) }
     end
 
     # Obtain a list of relationships starting at the passed resource
     # @param [Resource] resource One end of the relationship
+    # @param [String] named Name of the relationship
     # @return [Array<Relationship>] List of relationships
-    def list_relationships(resource)
-      the_feed = hawk_escape resource.feed
-      the_id = hawk_escape resource.id
-
-      ret = http_get('/feeds/' + the_feed + '/resources/' + the_id + '/relationships')
-      val = []
-      ret.each { |r| val.push(Relationship.new(r)) }
-      val
-    rescue
-      []
+    def list_relationships(resource, named = nil)
+      query = named.nil? ? '' : (generate_query_params named: named)
+      ret = http_get("/path#{resource.path}/relationships#{query}")
+      ret.map { |r| Relationship.new(r) }
     end
 
     # Obtain a list of relationships for the passed feed
     # @param [String] feed_id Id of the feed
+    # @param [String] named Name of the relationship
     # @return [Array<Relationship>] List of relationships
-    def list_relationships_for_feed(feed_id)
+    def list_relationships_for_feed(feed_id, named = nil)
       the_feed = hawk_escape feed_id
-      ret = http_get('/feeds/' + the_feed + '/relationships')
-      val = []
-      ret.each { |r| val.push(Relationship.new(r)) }
-      val
+      query = named.nil? ? '' : (generate_query_params named: named)
+      ret = http_get("/feeds/#{the_feed}/relationships#{query}")
+      ret.map { |r| Relationship.new(r) }
     rescue
       []
     end
 
+    # Retrieve a single entity from inventory by its canonical path
+    # @param [String] path canonical path of the entity
+    # @return inventory entity
+    def get_entity(path)
+      http_get("path#{path}")
```

Comment:
this works, I'll add to the docs that it only accept the absolute path (i.e. starting with /)
or it can actually accept the CanonicalPath instance to provide some kind of "type safety"